pvlib/mlfm.py

```diff
@@ -0,0 +1,804 @@
+
+    See also
+    --------
+    mlfm_norm_to_stack
+    '''
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        raise ImportError('plt_mlfm_stack requires matplotlib')
+
+    stack6 = ['i_sc', 'r_sc', 'i_ff', 'i_v', 'v_ff', 'r_oc', 'v_oc']
+    stack4 = ['i_sc', 'i_mp', 'i_v', 'v_mp', 'v_oc']
+
+    if all([c in dstack.columns for c in stack6]):
+        # data order from bottom to top
+        ydata = [dnorm['pr_dc'] + (dstack['i_sc'] * (is_i_sc_self_ref)),
+                 dstack['temp_module_corr'] * (is_v_oc_temp_module_corr),
+                 dstack['v_oc'] - (
+                     dstack['temp_module_corr'] * (is_v_oc_temp_module_corr)),
+                 dstack['r_oc'],
+                 dstack['v_ff'],
+                 dstack['i_v'],
+                 dstack['i_ff'],
+                 dstack['r_sc'],
+                 dstack['i_sc'] * (not is_i_sc_self_ref)]
+        labels = [
+            'pr_dc',
+            'stack_t_mod',
+            'stack_v_oc',
+            'stack_r_oc',
+            'stack_v_ff',
+            '- - -',
+            'stack_i_ff',
+            'stack_r_sc',
+            'stack_i_sc']
+        color_map = [
+            'white',  # colour to bottom of graph
+            clr['temp_module'],
+            clr['v_oc'],
+            clr['r_oc'],
+            clr['v_ff'],
+            clr['i_v'],
+            clr['i_ff'],
+            clr['r_sc'],
+            clr['i_sc']]
+
+    if all([c in dstack.columns for c in stack4]):
+        # data order from bottom to top
+        ydata = [dnorm['pr_dc'] + (dstack['i_sc'] * (is_i_sc_self_ref)),
+                 dstack['temp_module_corr'] * (is_v_oc_temp_module_corr),
+                 dstack['v_oc'] - (
+                     dstack['temp_module_corr'] * (is_v_oc_temp_module_corr)),
+                 dstack['v_mp'],
+                 dstack['i_v'],
+                 dstack['i_mp'],
+                 dstack['i_sc'] * (not is_i_sc_self_ref)]
+        labels = [
+            'pr_dc',
+            'stack_t_mod',
+            'stack_v_oc',
+            'stack_v_mp',
+            '- - -',
+            'stack_i_mp',
+            'stack_i_sc']
+        color_map = [
+            'white',  # colour to bottom of graph
+            clr['temp_module'],
+            clr['v_oc'],
+            clr['v_mp'],
+            clr['i_v'],
+            clr['i_mp'],
+            clr['i_sc']]
+
+    # offset legend right, use ~1.2
+    bbox = 1.2
+
+    # select x axis usually date_time
+    xdata = dmeas.index.values
+    fig, ax1 = plt.subplots()
+
+    ax1.set_title(title)
+
+    # plot stack in order bottom to top,
+    # allowing self_ref and temp_module corrections
+    ax1.stackplot(xdata, *tuple(ydata), labels=labels, colors=color_map)
+
+    ax1.axhline(y=1/fill_factor, c='grey', lw=3)  # show initial 1/FF
+    ax1.axhline(y=1, c='grey', lw=3)  # show 100% line
+    ax1.set_ylabel('stacked mlfm losses')
+
+    # find number of x date values
```

Comment:
Fancy date labels could be a separate utility function reusable elsewhere and by others.

Reply:
Am very happy for that to happen as making date axes legible with lots of data isn't easy. Will look at ways to do it but am open to suggestions.

Reply:
Not sure if this helps: https://matplotlib.org/stable/gallery/text_labels_and_annotations/date.html